front/components/actions/mcp/ToolsList.tsx

```diff
@@ -88,20 +74,23 @@ export function ToolsList({
     permission: CustomRemoteMCPToolStakeLevelType | "never_ask",
     enabled: boolean
   ) => {
-    void updateToolSettings({
-      toolName: name,
-      permission,
-      enabled,
-    });
+    form.setValue(
+      `toolSettings.${name}`,
+      { permission, enabled },
+      {
+        shouldDirty: true,
+      }
+    );
```

Comment:
Instead we usually try to stick to `useController`, it also let's you retrieve the field value directly

Reply:
In general when possible to use it's cleaner to default to it